rs/sns/governance/src/governance.rs

```diff
@@ -4754,6 +4841,96 @@ impl Governance {
             }
         }
     }
+    /// Checks if an automatic upgrade is needed and initiates it.
+    /// An automatic upgrade is needed if `target_version` is set to a future version on the upgrade path
+    async fn initiate_upgrade_if_behind_target_version(&mut self) {
+        if self.check_no_upgrades_in_progress().is_err() {
+            // An upgrade is already in progress
+            return;
+        }
+
+        let Some(deployed_version) = self.proto.deployed_version.clone() else {
+            return;
+        };
+
+        let Some(target_version) = self.proto.target_version.clone() else {
+            return;
+        };
+
+        let Some(CachedUpgradeSteps {
+            upgrade_steps: Some(Versions {
+                versions: upgrade_steps,
+            }),
+            ..
+        }) = &self.proto.cached_upgrade_steps
+        else {
+            log!(ERROR, "Cached upgrade steps set to None");
+            return;
+        };
+
+        // Find the current position of the deployed version
+        let Some(deployed_position) = upgrade_steps.iter().position(|v| v == &deployed_version)
+        else {
+            log!(ERROR, "Deployed version is not on the upgrade path");
```

Comment:
```suggestion
            log!(ERROR, "Deployed version {} is not on the upgrade path {}", deployed_version.human_readable(), );
```

```
impl Version {
  pub fn human_readable(&self) -> String {
        format!(
            "SnsVersion {{ root:{}, governance:{}, swap:{}, index:{}, ledger:{}, archive:{} }}",
            hash_to_hex_string(self.root_wasm_hash),
            hash_to_hex_string(self.governance_wasm_hash),
            hash_to_hex_string(self.swap_wasm_hash),
            hash_to_hex_string(self.index_wasm_hash),
            hash_to_hex_string(self.ledger_wasm_hash),
            hash_to_hex_string(self.archive_wasm_hash),
        )
    }
}
```

Or implement `Display for Version`.

Reply:
Instead of logging, this information should in added to the upgrade journal.